core/src/main/scala/kafka/coordinator/group/DelayedHeartbeat.scala

```diff
@@ -25,12 +25,13 @@ import kafka.server.DelayedOperation
  */
 private[group] class DelayedHeartbeat(coordinator: GroupCoordinator,
                                       group: GroupMetadata,
+                                      pendingMemberId: String,
                                       member: MemberMetadata,
```

Comment:
Since this can be null, would it be better if we used an `Option<MemberMetadata`>?

Reply:
I will be removing the member from heartbeat purgatory.